aspnetcore/razor-components/components.md

```diff
@@ -761,3 +761,61 @@ The time is 10/04/2018 01:26:52.
 
 Your pet's name is Rex.
 ```
+
+## Manual RenderTreeBuilder logic
+
+`Microsoft.AspNetCore.Components.RenderTree` provides methods for manipulating components and elements, including building components manually in C# code.
+
+> [!NOTE]
+> Use of `RenderTreeBuilder` to create components is an advanced scenario. A malformed component (for example, an unclosed markup tag) can result in undefined behavior.
+
+When calling `RenderTreeBuilder` methods, sequence numbers are source code line numbers. The Razor Components difference algorithm relies on the sequence numbers corresponding to distinct lines of code, not distinct call invocations. When creating a component with `RenderTreeBuilder`, hardcode the sequence number. Using a calculation or counter to generate the sequence number can lead to poor performance.
```

Comment:
I think it would read better if this paragraph was moved to the end of the paragraph before the example that shows the sequence numbers

Reply:
I think I'll need to add something for the Pet Details component, too. It's somewhat out-of-context anyway ... not clear what purpose it serves.